### DISCUSSION TITLE
VGABIOS-lgpl source code

### DISCUSSION THREAD

**Discussion by Vort** [bochs-emu/Bochs#168] [Q&A] [answered]
Are there reasons _(apart from lack of time)_ to not include source code for `VGABIOS-lgpl-latest` either in this repository (somewhere near regular bios) or in separate repository of `bochs-emu` organization?

**Comment by vruppert** [comment]
Two people started the LGPL'd VGABIOS project long time ago on Savannah and it became the default VGABIOS for Bochs and Qemu. Then I joined this project and some years later the other guys disappeared. I think it could be added as a separate project under bochs-emu if possible.

**Reply by Vort** [reply]
I looked at QEMU repository and it seems they use seabios now:
https://github.com/qemu/seabios/tree/master/vgasrc
So I expect moving (or copying) VGABIOS-lgpl code to this place won't make problems for them.
But maybe someone else use it, I am not 100% sure.

Separate repositories can be linked with help of submodules feature of git.
However including code directly looks more straightforward.

**Reply by stlintel** [reply]
I believe Bochs should also move to seabios as primary one or at least have it supported in the same level as Bochs BIOS.
Also having support for UEFI Bios would be really nice.
QEMU has 'EFI BIOS for QEMU":
[efi-bios.tar.gz](https://github.com/bochs-emu/Bochs/files/13663747/efi-bios.tar.gz)
and also OVMF (Open Virtual Machine Firmware)

Did  anybody tried too boot any of them ?

- Recent seabios (bochs worked well with seabios 4-years ago, not sure it still does with recent one) ?
- OVMF (Open Virtual Machine Firmware) ?

**Reply by Vort** [reply]
> Did anybody tried too boot any of them ?

[vgabios.zip](https://github.com/bochs-emu/Bochs/files/13670075/vgabios.zip) mostly works, but looks like it have problems with changing resolution after Reset.
[OVMF-pure-efi.zip](https://github.com/bochs-emu/Bochs/files/13670086/OVMF-pure-efi.zip) shows shell, but I wasn't able to boot any OS with it.
[RELEASEX64_OVMF.zip](https://github.com/bochs-emu/Bochs/files/13670099/RELEASEX64_OVMF.zip) fails to load at all, because of `ROM image too large`.

**Reply by vruppert** [reply]
I have now added the LGPL'd VGABIOS sources at https://github.com/vruppert/VGABIOS. I wanted to place it under "bochs-emu", but I don't know  how to do it. I could also update the SeaBIOS binaries in the "bios" folder to current version 1.16.3.

**Reply by Vort** [reply]
> I wanted to place it under "bochs-emu", but I don't know how to do it

Maybe this button allows to do this?
![image](https://github.com/bochs-emu/Bochs/assets/1242858/6ddcb919-09e8-4916-853c-308c8de0e5b1)

**Reply by vruppert** [reply]
Before importing I forgot to change the owner. Now I missed something else. The repository is now present under "bochs-emu", but it's private. I found out how to change this, but it doesn't seem to work.

**Reply by sskras** [reply]
> I have now added the LGPL'd VGABIOS sources at https://github.com/vruppert/VGABIOS. I wanted to place it under "bochs-emu", but I don't know how to do it. I could also update the SeaBIOS binaries in the "bios" folder to current version 1.16.3.

I believe you should do that through the settings tab –> the **Danger Zone** section –> the **Transfer ownership** item:
https://github.com/vruppert/VGABIOS/transfer

Then proceed to **New owner** *, eg. one of my repos:

<img width="1006" height="748" alt="image" src="https://github.com/user-attachments/assets/29c6c8d9-7be1-4733-a505-d06eb32f4afe" />

**Reply by sskras** [reply]
PS. Ah, please excuse me – this was already solved ago.

**Comment by stlintel** [comment]
Hi Volker,

What version control it was before ?
I have converted SVN repository of Bochs to git and also did the same for sfsite.
I believe It is possible to convert it to git and preserve all revision history.
Would you like me to try that ?

For me even https://github.com/vruppert/VGABIOS is 404 Error ...

Thanks
Stanislav

**Reply by vruppert** [reply]
It was originally CVS, but I switched to SVN after we did it with Bochs. Now I managed to import it under "bochs-emu" and deleted the version on my personal page, but it's private now and the procedure to make it public has no effect.

**Reply by vruppert** [reply, marked as answer]
Update: Now I managed to make the VGABIOS repository public. I had to confirm the final ask dialog twice to make it work.

**Reply by hangshan-boop** [reply]
Mr vruppert , I need  bochs-emu under BOIS that complete source code(not binary). Please can you give me？

**Comment by vruppert** [comment]
Source code is available at https://github.com/bochs-emu/VGABIOS.

**Comment by hangshan-boop** [comment]
These bin file code that source code as C, C++ or assembly format. Please can you give me?




Best regards!
jianli







---Original---
From: "Volker ***@***.***&gt;
Date: Fri, Feb 7, 2025 20:34 PM
To: ***@***.***&gt;;
Cc: ***@***.******@***.***&gt;;
Subject: Re: [bochs-emu/Bochs] VGABIOS-lgpl source code (Discussion #168)




 
Source code is available at https://github.com/bochs-emu/VGABIOS.
